src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs

```diff
@@ -78,6 +83,29 @@ internal virtual IList<string> FindPotentialSdkResolvers(string rootFolder, Elem
             return assembliesList;
         }
 
+        private DirectoryInfo[] GetSubfolders(string rootFolder, string additionalResolversFolder)
+        {
+            DirectoryInfo[] subfolders = null;
+            var rootFolderInfo = new DirectoryInfo(rootFolder);
+            if (rootFolderInfo.Exists)
+            {
+                subfolders = rootFolderInfo.GetDirectories();
+            }
+
+            if (additionalResolversFolder != null)
+            {
+                var resolversDirInfo = new DirectoryInfo(additionalResolversFolder);
+                if (resolversDirInfo.Exists)
+                {
+                    var overrideFolders = resolversDirInfo.GetDirectories();
+                    var overriddenFolders = subfolders.Where(subfolder => overrideFolders.Any(overrideFolder => overrideFolder.Name.Equals(subfolder.Name)));
```

Comment:
This would be much prettier if you made an EqualityComparer for this, wrapped it in a HashSet, added all the elements of overrideFolders to that HashSet, and did:
overrideFolders.Concat(subfolders.Where(e => set.Add(e)));

This is also very inefficient as-is. Make sure to use StringComparer.OrdinalIgnoreCase if on a case-insensitive OS.

Reply:
I'm not fully convinced that this would be any cleaner, it seems like a waste to allocate a hashset and create a comparer when there are other alternatives.

Reply:
Note that this code should only be running when the test hook is enabled, so the perf is not as critical.

Generally a HashSet is probably better, but I'm not really sure if there are usually only 3 or so items in the sets.

Reply:
That's fine with me, I've updated the PR.